### PR TITLE
Include sortBy parameter in BrAPI field import

### PR DESCRIPTION
**Pull request by bellerbrock**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

Adds a spinner to choose sortBy parameter in BrAPI field imports. Sorting by studyDbId rather then studyName allows faster access to most recently created fields in a many-paged list result.

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
sortBy parameter added to BrAPI field import
```

### CODE REVIEW

**Comment by trife** [comment]
I'm not sure if this is working for me. I was testing with cassavabase but the order of returned studies didn't change regardless of the choice I selected.

**Comment by bellerbrock** [comment]
Try with sweetpotatobase.org - the sortBy parameter has been part of BrAPI spec, but breedbase only just added it and I think not all their sites are updated with the latest release.

But come to think of it maybe there should be a way to test if the server has sortBy implemented, and hide the option if it isn't going to work.

File: app/src/main/java/com/fieldbook/tracker/activities/brapi/BrapiActivity.java [hunk 1]
```diff
@@ -107,7 +114,25 @@ private void loadToolbar() {
...
+        Spinner sortBySpinner = findViewById(R.id.sortBySpinner);
+        ArrayAdapter<CharSequence> sortByAdapter = ArrayAdapter.createFromResource(this,
+                R.array.brapi_study_sort_by_options, android.R.layout.simple_spinner_item);
+        sortByAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        sortBySpinner.setAdapter(sortByAdapter);
+        sortBySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                sortBy = parent.getItemAtPosition(position).toString();
+                Toast.makeText(BrapiActivity.this, "New sortBy option selected: " + sortBy, Toast.LENGTH_SHORT).show();
```
Comment (chaneylc) [inline app/src/main/java/com/fieldbook/tracker/activities/brapi/BrapiActivity.java:128]: Your toast should use a string resource.

**Review by chaneylc — CHANGES_REQUESTED** [review]
My only other comment is possibly renaming the sortBy spinner options to be something more readable, but I get if we want to keep those completely brapi oriented. I thought things like: Study Name, Study ID, Trial Name, ... might be better.

File: app/src/main/java/com/fieldbook/tracker/activities/brapi/BrapiActivity.java (hunk 1 above)
Comment (bellerbrock) [reply app/src/main/java/com/fieldbook/tracker/activities/brapi/BrapiActivity.java]: Good point. Was only needed for debugging anyway, so just removed it

**Comment by trife** [comment]
@bellerbrock go ahead  Replace sort options with user-friendly strings. I'm also not sure if sorting by DbIds makes much sense since they're usually arbitrary. Planting/creation date or a similar timestamped field would be a good addition.

**Comment by bellerbrock** [comment]
Here's some [more info from the BrAPI spec](https://github.com/plantbreeding/BrAPI/blob/eed6a7a2dc4dc53d4a93743c5307fce47f43e4ac/Specification/BrAPI-Core/Studies/Studies_GET_POST.yaml#L56-L82). Users can choose what they want to sort by from those options, but no date fields in there.

Which makes some sense to me because not all studies will have associated dates. But in all cases I've seen (Breedbase,T3,BIMS) studyDbId is an incrementing number that serves as the primary key in the table that stores studies, so it will sort in the same order as a create date would.

I'll switch to user-friendly strings and convert this to a draft for now. I want to talk to Pete and make a few other refinements before merging. like including sortOrder and only providing the sort option if sorting is actually implemented by the server.

**Comment by trife** [comment]
@bellerbrock this can be closed, right? It looked like this was merged with the datagrid sorting

**Comment by bellerbrock** [comment]
@trife Ah, I see the confusion since they both refer to 'sort order' implementation, but no the functionality is distinct.

This PR adds sortBy and sortOrder parameters to the study GET request, while the datagrid PR added an additional attribute spinner to the BrAPI field import's preview dialog to ensure that the study is saved in Field Book with a sort order (important for cases where the obs units were being imported in random order).

**Comment by bellerbrock** [comment]
Closing this, this and many more improvements are already handled by #1073

**Pull request closed without merging**